gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-sync/src/main/java/io/gravitee/gateway/services/sync/process/repository/synchronizer/license/LicenseSynchronizer.java

```diff
@@ -0,0 +1,105 @@
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author Guillaume LAMIRAND (guillaume.lamirand at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class LicenseSynchronizer implements RepositorySynchronizer {
+
+    private final LicenseFetcher licenseFetcher;
+    private final DeployerFactory deployerFactory;
+    private final ThreadPoolExecutor syncFetcherExecutor;
+    private final ThreadPoolExecutor syncDeployerExecutor;
+
+    @Override
+    public Completable synchronize(final Long from, final Long to, final List<String> environments) {
+        AtomicLong launchTime = new AtomicLong();
+        return licenseFetcher
+            .fetchLatest(from, to)
+            .subscribeOn(Schedulers.from(syncFetcherExecutor))
+            .rebatchRequests(syncFetcherExecutor.getMaximumPoolSize())
+            .flatMap(licenses -> Flowable.fromStream(licenses.stream().map(this::prepareForDeployment)))
+            // per deployable
+            .compose(upstream -> {
+                LicenseDeployer licenseDeployer = deployerFactory.createLicenseDeployer();
+                return upstream
+                    .parallel(syncDeployerExecutor.getMaximumPoolSize())
+                    .runOn(Schedulers.from(syncDeployerExecutor))
+                    .flatMap(deployable -> deploy(licenseDeployer, deployable))
+                    .sequential(1); // TODO CHECK THIS
+            })
+            .count()
+            .doOnSubscribe(disposable -> launchTime.set(Instant.now().toEpochMilli()))
+            .doOnSuccess(count -> {
+                String logMsg = String.format(
+                    "%s dictionaries synchronized in %sms",
+                    count,
+                    (System.currentTimeMillis() - launchTime.get())
+                );
+                if (from == -1) {
+                    log.info(logMsg);
+                } else {
+                    log.debug(logMsg);
+                }
+            })
+            .ignoreElement();
+    }
+
+    private LicenseDeployable prepareForDeployment(final License organizationLicense) {
+        return LicenseDeployable
+            .builder()
+            .id(organizationLicense.getReferenceId())
+            .license(organizationLicense.getLicense())
+            .syncAction(SyncAction.DEPLOY)
+            .build();
+    }
+
+    private static Flowable<LicenseDeployable> deploy(final LicenseDeployer licenseDeployer, final LicenseDeployable deployable) {
+        return licenseDeployer
+            .deploy(deployable)
+            .andThen(licenseDeployer.doAfterDeployment(deployable))
+            .andThen(Flowable.just(deployable))
+            .onErrorResumeNext(throwable -> {
+                log.error(throwable.getMessage(), throwable);
+                return Flowable.empty();
+            });
+    }
+
+    @Override
+    public int order() {
+        return 50;
```

Comment:
The order should be revised to make sure licenses are loaded before anything else. API deployment will depend on the organization's license to check features.

Reply:
API synchronizer is 1. Should I set this one to 0, or should I use 1 and change order for API ?

Reply:
Both options sound valid